src/player.js

```diff
@@ -1183,34 +1185,32 @@ export default class Player extends FakeEventTarget {
    * @returns {void}
    */
   notifyExitFullscreen(): void {
-    if (this._fullscreen) {
-      this._fullscreen = false;
-      this.dispatchEvent(new FakeEvent(CustomEventType.EXIT_FULLSCREEN));
-    }
+    this.dispatchEvent(new FakeEvent(CustomEventType.EXIT_FULLSCREEN));
   }
 
   /**
    * Request the player to enter fullscreen.
    * @public
+   * @param {HTMLElement} fullScreenElement - set html element to full screen
    * @returns {void}
    */
-  enterFullscreen(): void {
-    if (!this._fullscreen) {
-      if (this.isInPictureInPicture()) {
-        this.exitPictureInPicture();
-      }
-      this.dispatchEvent(new FakeEvent(CustomEventType.REQUESTED_ENTER_FULLSCREEN));
+  enterFullscreen(fullScreenElement: ?HTMLElement): void {
+    //check for fullscreen or IOS device with flag of kaltura full screen - full screen by style not native
+    if (!this.isFullscreen() || (this._getInBrowserFullscreenForIOS() && this._env.os.name === 'iOS')) {
```

Comment:
move this logic to controller. 
fix the condition in ios to look at the current mode not the config